File: javascript/7kyu/Sum of Odd Cubed Numbers.js
// https://www.codewars.com/kata/580dda86c40fa6c45f00028a

// Find the sum of the odd numbers within an array, after cubing the initial integers. The function should return undefined if any of the values aren't numbers.

function cubeOdd(arr) {
    var sum = 0;
    for (var i = 0; i <arr.length; i++){
      var cubed = (arr[i]*arr[i]*arr[i]);
      if (isNaN(cubed)) return undefined;
      if(cubed % 2 != 0 )sum += cubed;
    }
    return sum;
  }

File: java/7kyu/Sum of Odd Cubed Numbers.java
// https://www.codewars.com/kata/580dda86c40fa6c45f00028a

// Find the sum of the odd numbers within an array, after cubing the initial integers. The function should return undefined if any of the values aren't numbers.

public class FindOddCubes {
    public static int cubeOdd(int arr[]) {
        int result = 0;
        for(int i : arr) {
            if(i % 2 != 0)  result += Math.pow(i, 3);   
        }
        return result;
    }
}